Core/MMKV.cpp

```diff
@@ -595,7 +602,9 @@ bool MMKV::set(const char *value, MMKVKey_t key, uint32_t expireDuration) {
     }
     if (likely(!m_enableKeyExpire)) {
         assert(expireDuration == ExpireNever && "setting expire duration without calling enableAutoKeyExpire() first");
-        return setDataForKey(MMBuffer((void *) value, strlen(value), MMBufferNoCopy), key, true);
+
+        auto data = std::move(MMBuffer((void *) value, strlen(value), MMBufferNoCopy));
+        return setDataForKey(std::move(data), key, true);
```

Comment:
Split into a temp object and call `std::move()` is pointless. A temporary object is already an R-value.